pyramid_chameleon/precompile.py

```diff
@@ -16,23 +16,26 @@
 from pyramid_chameleon.zpt import PyramidPageTemplateFile
 
 def _compile_one(args):
-    fullpath = args[0]
-    template_factory = args[1]
-    fail_fast = args[2]
+    fullpath, template_factory, fail_fast, cache_dir = (
+        args[0],
+        args[1],
+        args[2],
+        args[3],
+        )
     try:
-        compile_one(fullpath, template_factory)
+        compile_one(fullpath, cache_dir, template_factory)
     except KeyboardInterrupt:
         return dict(path=fullpath, success=False)
-    except:
+    except Exception as e:
         if fail_fast:
             raise
-        logging.warn('Failed to compile: %s' % fullpath)
+        logging.error('Failed to compile: %s' % fullpath, exc_info=e)
         return dict(path=fullpath, success=False)
     logging.debug('Compiled: %s' % fullpath)
     return dict(path=fullpath, success=True)
 
-def compile_one(fullpath, template_factory=PyramidPageTemplateFile):
-    assert chameleon.config.CACHE_DIRECTORY is not None
+def compile_one(fullpath, cache_dir, template_factory=PyramidPageTemplateFile):
+    chameleon.config.CACHE_DIRECTORY = cache_dir
```

Comment:
Kind of narsty to be scribbling on chameleon's globals, but then again, it is kind of narsty that it pulls envvars into its own globals.  Might we be better off doing the equivalent of `mock.patch` to that module?

Also, is `compile_one` an API?  Because if so, we are breaking compat here.

*/me looks*
`pyramid_chameleon.precompile` is not included in `docs/api.rst`, so we are likely safe.

Reply:
Yeah the only API is the name of the script and the arguments it takes.